src/components/Search/index.tsx

```diff
@@ -159,9 +237,80 @@ const Search = ({ dispatch, hasBorder, content }: { dispatch: AppDispatch; hasBo
           }
         }}
       />
+
       {hasBorder && <SearchIconButton greenIcon />}
+
+      {searchSuggestions.length > 0 && searchValue ? (
+        <SuggestionsDropdown
+          suggestions={searchSuggestions}
+          onSuggestionSelect={onSuggestionSelect}
+          expandSuggestions={expandSuggestions}
+          setExpandSuggestions={setExpandSuggestions}
+        />
+      ) : null}
     </SearchPanel>
   )
 }
 
+interface SearchSuggestion {
+  value?: string
+  balance?: number
+  path?: string
+  type: string
+}
+
+const generateSuggestionValue = (suggestion: SearchSuggestion) => {
+  if (isMobile() && suggestion.type !== 'Block Height' && suggestion.value) {
+    return parseLongAddressHashMobile(suggestion.value)
+  }
+  return suggestion.value
+}
+
+const generateFormattedBalance = (suggestion: SearchSuggestion) => {
+  if (suggestion.balance) {
+    return `${(suggestion.balance / 10 ** 8).toFixed(3)} ckb`
+  }
+  return ''
+}
+
+const SuggestionsDropdown = ({
+  suggestions = [],
+  onSuggestionSelect,
+  expandSuggestions,
+  setExpandSuggestions,
+}: {
+  suggestions: SearchSuggestion[]
+  onSuggestionSelect: any
+  expandSuggestions: boolean
+  setExpandSuggestions: any
+}) => (
+  <SuggestionsPanel>
+    <SuggestionHeading>{suggestions.length > 0 && suggestions[0].type}</SuggestionHeading>
+    {suggestions
+      .slice(0, expandSuggestions ? suggestions.length : 3)
+      .filter(suggestion => suggestion.type !== 'Error')
```

Comment:
Wouldn't it be slightly better to have the `slice` come first because if `suggestions` is an array of 10 block heights, we can slice that list down to 3 before `filter`. 

Either way I think the time complexity is the same so there won't be a big performance difference!

Reply:
The point is not related to performance.

When there are suggestions with the type of `Error` in the top 3, `slice`+`filter` displays less than 3 items in the list and leave other items with index > 2 in the `more items`

`| Error | Suggestion 1 | Suggestion 2 | Suggestion 3 |` will be 
`| Suggestion 1 | Suggestion 2| More... | Suggestion 3 |` 
but the better list is ` | Suggestion 1 | Suggestion 2 | Suggestion 3 |`

Reply:
It's OK cuz the list including `Error` suggestion is limited to have only 1 item
https://github.com/nervosnetwork/ckb-explorer-frontend/pull/253/files#diff-78a58c8b7a6035584617b29d129ea101R163